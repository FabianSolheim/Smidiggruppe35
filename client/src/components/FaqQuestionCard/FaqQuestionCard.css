.accordion {
    max-width: 450px;
    margin: auto;
}
.item {
    margin-bottom: 20px;
    padding-bottom: 5px;
    border-bottom: 1px solid #034f7a;
}

.title {
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;
}

.content {
    max-height: 0;
    overflow: hidden;
    transition: all 0.3s cubic-bezier(0,1,0,1);
}

.content.show {
    height: auto;
    max-height: 9999px;
    transition: all 0.3s cubic-bezier(1,0,1,0);
}