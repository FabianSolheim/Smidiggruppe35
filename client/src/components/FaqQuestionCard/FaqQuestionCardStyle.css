.card-container{
    display: inline-block;
    inline-size: 300px;
    cursor: pointer;
    overflow-wrap: break-word;


}
.underline{
    
    height: 1px;
    width: 100%;
    background-color: #034F7A;
}
.hidden-text-container{
    position: relative;
    top: -1rem;
    overflow-wrap: break-word;
}
