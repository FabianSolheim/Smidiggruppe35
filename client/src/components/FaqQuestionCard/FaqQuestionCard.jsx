import React from "react";
import { useState } from "react";
import "./FaqQuestionCard.css";
import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
import RemoveCircleOutlineIcon from "@mui/icons-material/RemoveCircleOutline";

export function FaqQuestionCard({ faqs: { _id, question, answer }, index }) {
  const [selected, setSelected] = useState(null);
  const toggle = (index) => {
    if (selected === index) {
      return setSelected(null);
    }
    setSelected(index);
    console.log(selected + index);
  };
  return (
    <div className="accordion">
      <div className="item">
        <div className="title" onClick={() => toggle(index)}>
          <div>
            <strong>
              {index + 1}. {question}
            </strong>
          </div>
          <span>
            {selected === index ? (
              <RemoveCircleOutlineIcon
                style={{ color: "#034F7A", fontSize: "25px" }}
              />
            ) : (
              <AddCircleOutlineIcon
                style={{ color: "#034F7A", fontSize: "25px" }}
              />
            )}
          </span>
        </div>
        <div className={selected === index ? "content show" : "content"}>
          <div>{answer}</div>
        </div>
      </div>
    </div>
  );
}
