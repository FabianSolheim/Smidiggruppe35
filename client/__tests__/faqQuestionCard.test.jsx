import React from "react";
import ReactDOM from "react-dom";
import { FaqQuestionCard } from "../src/components/FaqQuestionCard/FaqQuestionCard";

describe("FaqQuestionCard test", () => {
  it("shows FaqQuestionCard", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<div>{FaqQuestionCard}</div>, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
