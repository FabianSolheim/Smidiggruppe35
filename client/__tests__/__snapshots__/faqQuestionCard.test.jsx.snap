// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqQuestionCard test shows FaqQuestionCard 1`] = `"<div class=\\"accordion\\"><div class=\\"item\\"><div class=\\"title\\" id=\\"faqBtn\\"><div><strong>1. testQuestion</strong></div><span><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"AddCircleOutlineIcon\\" style=\\"color: rgb(73, 3, 116); font-size: 25px;\\"><path d=\\"M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z\\"></path></svg></span></div><div class=\\"content\\"><div>testAnswer</div></div></div></div>"`;
